packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -259,7 +260,11 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
           // its last updated timestamp to force the browser to fetch the most
           // up-to-date version of this module.
           try {
-            const depModule = await moduleGraph.ensureEntryFromUrl(url, ssr)
```

Comment:
This is where the resolved metadata was being lost

Reply:
Indeed, it was getting lost here because the `url`, passed to it was already query-trimmed.  So I agree with your change here, and yes, I think it means you can revert this change in `moduleGraph.ts`.

```diff
       fileMappedModules.add(mod)
+    } else if (mod.id !== resolvedId) {
+      this.idToModuleMap.set(resolvedId, mod)
+      if (meta) mod.meta = { ...mod.meta, ...meta }
     }
     return mo
```

*But should you?* Is not that change conceptually correct in the protocol of `ensureEntryFromResolved`? Depends on what one takes to be "key" inside `resolvedUrl` triplet.    If you say it's only the URL, you're right to revert it.  If you assume its the URL/id combination, you're not.

Reply:
The rationale for removing that block goes like this:  
The resolution of a module URL is assumed to be idempotent, which means the `resolvedId` and `meta` values are not expected to change between `resolveId` invocations. This assumption is made clear by the `ModuleNode` object having only a single `id` string, rather than an array of them.

Reply:
> The resolution of a module URL is assumed to be idempotent, which means the resolvedId and meta values are not expected to change between resolveId invocations. 

Makes sense. And you now made it so that that expectation is fulfilled where it wasn't. However, they are still three arguments to the function that will sometimes ignore the latter two if they vary.  This means that even with this new somewhat cleaner helper, it could perhaps be even cleaner and take only one argument.  Not for this PR though, and maybe just stating what you said in a comment/docstring would probably be more than enough.